src/abe/dippe.c

```diff
@@ -0,0 +1,613 @@
+
+#include <math.h>
+#include <string.h>
+#include <amcl/big_256_56.h>
+#include <amcl/pair_BN254.h>
+
+#include "cifer/internal/common.h"
+#include "cifer/internal/big.h"
+#include "cifer/sample/uniform.h"
+#include "cifer/abe/dippe.h"
+
+
+/**
+ * Helper function that returns the number of digits for an int
+ *
+ * @param i The integer to determine the digit count from
+ * @return The digit count
+ */
+int cfe_dippe_int_len(int i) {
+    return i ? (((int)log10((double)i))+1) : 1;
+}
+
+/**
+ * Hashes the given buffer to an element of G2
+ *
+ * @param out A pointer to a ECP2_BN254 struct; The resulting group element
+ * @param in_buf Buffer that is about to be hashed
+ * @param in_buf_size Size of the buffer
+ */
+void cfe_dippe_hash_G2(ECP2_BN254 *out, char* in_buf, int in_buf_size) {
+    // rounds
```

Comment:
Could you give some reference to where this way of hashing comes from?

Reply:
I don't really have a reference. It looked like the buffer that is about to be hashed must be of size `MODBYTES_256_56`. I thought it of a way to hash greater buffers, that may occur during KeyGen. Maybe there's a better way.

Reply:
In the meantime we a bit reorganized the library and we put all the implemented hashing functions in `internal/hash.c` file. There you can also find some helping functions such needed for hashing such as string concatenation and vector to string changer. Could you rebase your contribution and just use these functions (`cfe_hash_G2`) instead of defining a new one.